src/sentry/replays/endpoints/project_replay_breadcrumbs.py

```diff
@@ -0,0 +1,246 @@
+        "statusCode": int,
+        "request_size": int,
+        "response_size": int,
+        # Navigation
+        "from": str,
+        "to": str,
+        # Click
+        "node_id": int,
+        "tag": str,
+        "text": str,
+        "is_dead": bool,
+        "is_rage": bool,
+        "selector": str,
+        "alt": str,
+        "aria_label": str,
+        "component_name": str,
+        "class": str,
+        "id": str,
+        "role": str,
+        "title": str,
+        "testid": str,
+    },
+    "default_limit": 50,
+    "default_offset": 0,
+    "extrapolation_mode": "none",
+}
+
+CATEGORY_ATTRIBUTE_MAP = {
+    "ui.click": [
+        "node_id",
+        "tag",
+        "text",
+        "is_dead",
+        "is_rage",
+        "selector",
+        "alt",
+        "aria_label",
+        "component_name",
+        "class",
+        "id",
+        "role",
+        "title",
+        "testid",
+    ],
+    "navigation": ["from", "to"],
+    "resource.xhr": [
+        "url",
+        "method",
+        "duration",
+        "statusCode",
+        "request_size",
+        "response_size",
+    ],
+    "resource.fetch": [
+        "url",
+        "method",
+        "duration",
+        "statusCode",
+        "request_size",
+        "response_size",
+    ],
+    "resource.script": [
+        "url",
+        "duration",
+        "size",
+        "statusCode",
+        "decodedBodySize",
+        "encodedBodySize",
+    ],
+    "resource.img": [
+        "url",
+        "duration",
+        "size",
+        "statusCode",
+        "decodedBodySize",
+        "encodedBodySize",
+    ],
+    "web-vital.cls": ["duration", "rating", "size", "value"],
+    "web-vital.fcp": ["duration", "rating", "size", "value"],
+    "web-vital.lcp": ["duration", "rating", "size", "value"],
+    "replay.hydrate-error": ["url"],
+    "replay.mutations": ["count"],
+    "sdk.options": [
+        "shouldRecordCanvas",
+        "useCompressionOption",
+        "blockAllMedia",
+        "maskAllText",
+        "maskAllInputs",
+        "useCompression",
+        "networkDetailHasUrls",
+        "networkCaptureBodies",
+        "networkRequestHasHeaders",
+        "networkResponseHasHeaders",
+        "sessionSampleRate",
+        "errorSampleRate",
+    ],
+    "memory": ["jsHeapSizeLimit", "totalJSHeapSize", "usedJSHeapSize", "endTimestamp", "duration"],
+}
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Replays"])
+class ProjectReplayBreadcrumbsEndpoint(ProjectEndpoint):
+    owner = ApiOwner.REPLAY
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, project: Project, replay_id: str) -> Response:
+        if not features.has(
+            "organizations:session-replay", project.organization, actor=request.user
+        ):
+            return Response(status=404)
+
+        filter_params = self.get_filter_params(request, project)
+        end, start = filter_params["end"], filter_params["start"]
```

Comment:
for a per-replay endpoint, i can see it might be useful to pass in absolute timestamps for start/end.... another version would also be to pass in offsets from the start of the replay

like the difference is: "give me all the breadcrumbs from 12:10am to 12:30am"
or "give me all the breadcrumbs in the first 20mins of the replay"

in the former i need to know the start time of the replay, so i'd need to load the record first. But in the latter i could send off the breadcrumb request earlier. 

could be future revision, nbd.

Reply:
I can default to the replay's started_at and finished_at values so you don't need to wait for the replay to load.